lib/ansible/modules/cloud/vmware/vmware_guest_facts.py

```diff
@@ -133,6 +133,20 @@
     properties: ["config.hardware.memoryMB", "guest.disk", "overallStatus"]
   delegate_to: localhost
   register: facts
+
+- name: Gather Managed object ID (moid) from a guest using the vSphere API output schema for REST Calls
+  vmware_guest_facts:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: no
+    datacenter: "{{ datacenter_name }}"
+    name: "{{ vm_name }}"
+    schema: "vsphere"
+    properties:
+      - _moId
```

Comment:
You might want to add an integration test for that.  According to the code upstream in pyvmomi, each object gets three properties:

https://github.com/vmware/pyvmomi/pull/732/files#diff-3d209b7e252f047d63b9b948cb74282cR282

       When a ManagedObject is encoded, it gains three properties:
            _vimid is the _moId (ex: 'vm-42')
            _vimref is the moRef (ex: 'vim.VirtualMachine:vm-42')
            _vimtype is the class name (ex: 'vim.VirtualMachine')

_moId is internal to pyvmomi.  _vimid would be visible in the json output.

Reply:
OK, I will add all three properties and testcase for that.